tests/test_server/test_lineage/test_dataset_lineage.py

```diff
@@ -260,6 +261,123 @@ async def test_get_dataset_lineage_with_granularity_operation(
     }
 
 
+async def test_get_dataset_lineage_with_granularity_dataset(
+    test_client: AsyncClient,
+    async_session: AsyncSession,
+    lineage_with_depth: LineageResult,
+    mocked_user: MockedUser,
+):
+    lineage = lineage_with_depth
+    # We need a middle dataset, which has inputs and outputs
+    lineage_dataset = lineage.datasets[1]
+    # If start dataset is d1 we should have this lineage: d0-d1-d2
+    datasets = lineage.datasets[:3]
+    datasests_ids = [dataset.id for dataset in datasets]
+
+    inputs = [input for input in lineage.inputs if input.dataset_id in datasests_ids]
+    assert inputs
+
+    outputs = [output for output in lineage.outputs if output.dataset_id in datasests_ids]
+    assert outputs
+
+    run_ids = {input.run_id for input in inputs} | {output.run_id for output in outputs}
+    runs = [run for run in lineage.runs if run.id in run_ids]
+    assert runs
+
+    datasets = await enrich_datasets(datasets, async_session)
+    runs = await enrich_runs(runs, async_session)
+    since = min(run.created_at for run in runs)
+
+    response = await test_client.get(
+        "v1/datasets/lineage",
+        headers={"Authorization": f"Bearer {mocked_user.access_token}"},
+        params={
+            "since": since.isoformat(),
+            "start_node_id": lineage_dataset.id,
+            "granularity": "DATASET",
+        },
+    )
+
+    assert response.status_code == HTTPStatus.OK, response.json()
+
+    inputs, outputs = inputs_and_outputs_to_json(merge_io_by_runs(inputs), merge_io_by_runs(outputs))
```

Comment:
`merge_io_by_runs` returns one Input for each Run. But `granularity=DATASET` should combine all `dataset1 -> dataset2` items into one Input.